src/meta/meta_ckpt.c

```diff
@@ -209,8 +210,32 @@ __ckpt_set(WT_SESSION_IMPL *session, const char *fname, const char *v, bool use_
     if (use_base && session->dhandle != NULL) {
         WT_ERR(__wt_scr_alloc(session, 0, &tmp));
         WT_ASSERT(session, strcmp(session->dhandle->name, fname) == 0);
+
+        /* Check the metadata is not corrupted. */
+        meta_base = session->dhandle->meta_base;
```

Comment:
Can session->dhandle->meta_base be NULL here ? If not, we may not need all those checks line 218 or we could simply play safe.

Reply:
Since we are in the `use_base` case, no I don't believe it can be NULL here. Also, given this new diagnostic code it is probably easier to read if you introduce a local `WT_DATA_HANDLE *dhandle` variable to get rid of so many indirections. They make it harder to read.